x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractor.java

```diff
@@ -64,6 +66,7 @@
     private boolean isCancelled;
     private boolean hasNext;
     private boolean searchHasShardFailure;
+    private final SetOnce<CrossValidationSplitter> crossValidationSplitter = new SetOnce<>();
```

Comment:
I needed to move the cross validation splitter in the extractor itself so that we skip the same docs while loading data to the c++ process as well as when we're joining results.

Reply:
This also means the cross validation package makes no sense being under `process`. I'll move it out in a follow up PR.